core/vm/instructions.go

```diff
@@ -191,50 +193,97 @@ func opGt(pc *uint64, evm *EVM, contract *Contract, memory *Memory, stack *Stack
 }
 
 func opSlt(pc *uint64, evm *EVM, contract *Contract, memory *Memory, stack *Stack) ([]byte, error) {
-	x, y := math.S256(stack.pop()), math.S256(stack.pop())
-	if x.Cmp(math.S256(y)) < 0 {
-		stack.push(evm.interpreter.intPool.get().SetUint64(1))
+	x, y := stack.pop(), stack.peek()
+
+	xSign := x.Cmp(tt255)
+	ySign := y.Cmp(tt255)
+
+	if xSign >= 0 {
+		//x negative
+		if ySign < 0 {
+			// y positive
+			y.SetUint64(1)
+			return nil, nil
+		}
+		// Both negative
+		if x.Cmp(y) < 0 {
+			y.SetUint64(1)
+		} else {
+			y.SetUint64(0)
+		}
 	} else {
-		stack.push(new(big.Int))
+		// x positive
+		if ySign >= 0 {
+			// y negative
+			y.SetUint64(0)
+			return nil, nil
+		}
+		// Both positive
+		if x.Cmp(y) >= 0 {
+			y.SetUint64(0)
+		} else {
+			y.SetUint64(1)
+		}
```

Comment:
Perhaps this would be cleaner? 

```go
switch {
	case xSign >= 0 && ySign < 0:
		y.SetUint64(1)

	case xSign < 0 && ySign >= 0:
		y.SetUint64(0)

	default:
		if x.Cmp(y) < 0 {
			y.SetUint64(1)
		} else {
			y.SetUint64(0)
		}
}
```